newIDE/app/src/ProjectCreation/ProjectPreCreationDialog.js

```diff
@@ -0,0 +1,107 @@
+// @flow
+import { Trans } from '@lingui/macro';
+import * as React from 'react';
+import Dialog from '../UI/Dialog';
+import FlatButton from '../UI/FlatButton';
+import RaisedButton from '../UI/RaisedButton';
+import { Column, Spacer } from '../UI/Grid';
+import LocalFolderPicker from '../UI/LocalFolderPicker';
+import TextField from '../UI/TextField';
+
+type Props = {|
+  open: boolean,
+  isOpening?: boolean,
+  onClose: () => void,
+  onCreate: () => void | Promise<void>,
+  outputPath?: string,
+  onChangeOutputPath?: (outputPath: string) => void,
+  projectName: string,
+  onChangeProjectName: (name: string) => void,
+|};
+
+const ProjectPreCreationDialog = ({
+  open,
+  isOpening,
+  onClose,
+  onCreate,
+  outputPath,
+  onChangeOutputPath,
+  projectName,
+  onChangeProjectName,
+}: Props): React.Node => {
+  const [projectNameError, setProjectNameError] = React.useState<?React.Node>(
+    null
+  );
+
+  let projectNameToDisplay =
+    projectName === null || projectName === undefined
+      ? 'Bonjour les cocos'
+      : projectName;
+
+  const onValidate = React.useCallback(
+    () => {
+      setProjectNameError(null);
+      if (!projectName) {
+        setProjectNameError(<Trans>Your project needs a name</Trans>);
```

Comment:
```suggestion
        setProjectNameError(<Trans>Your project needs a name.</Trans>);
```

Reply:
I think I'll use a completely different formulation, this one seems a bit weird.

Reply:
Yep thanks!